src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -112,6 +112,11 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
         {
             Debug.Assert(_builtApplication is not null);
 
+            if (Environment.IsDevelopment())
```

Comment:
This should use `context.Environment`

Reply:
I was thinking about that. `Environment` is the source of truth. That's what we pass to all the callbacks that take a `WebHostBuilderContext` or `HostBuilderContext`. It's also the `IWebHostEnvironment` service and soon to be the `IHostEnvironment`.

Of course, if we do everything right, these should be in synchronized. But if they don't match, `context.Environment` would be different from what nearly all of the rest of the environment-based logic in app uses.

Reply:
How cool is @Tratcher with doing this? He's been very unsupportive of using `IsDevelopment` in the framework. Also to add a data point, we often get users confused about why their static web assets don't work correctly when they pick a different env name (e.g. `Dev`)

Reply:
Concerns noted at https://github.com/dotnet/aspnetcore/issues/34548#issuecomment-883537130. At least in this case people can always call UseDeveloperExceptionPage themselves, this is only a fallback.

Reply:
I'm making a change where it will be, but it isn't right now and can't be source of truth at this point either. Code can mutate the environment later.

Reply:
```suggestion
            if (context.Environment.IsDevelopment())
```